python-avd/pyavd/_eos_designs/structured_config/base/snmp_server.py

```diff
@@ -187,35 +198,84 @@ def _snmp_hosts(self: AvdStructuredConfigBaseProtocol, snmp_settings: EosDesigns
         self.structured_config.snmp_server.hosts = snmp_hosts
 
     def _snmp_local_interfaces(self: AvdStructuredConfigBaseProtocol, source_interfaces_inputs: EosDesigns.SourceInterfaces.Snmp) -> None:
-        """Set local_interfaces if "source_interfaces.snmp" is set."""
+        """
+        Set local_interfaces from "source_interfaces.snmp".
+
+        TODO: AVD6.0 remove this method.
+        """
         if not source_interfaces_inputs:
-            # Empty dict or None
             return
-        local_interfaces = self._build_source_interfaces(
+
+        self.structured_config.snmp_server.local_interfaces = self._build_source_interfaces(
             source_interfaces_inputs.mgmt_interface,
             source_interfaces_inputs.inband_mgmt_interface,
             error_context="SNMP",
             output_type=EosCliConfigGen.SnmpServer.LocalInterfaces,
         )
-        self.structured_config.snmp_server.local_interfaces = local_interfaces
 
     def _snmp_vrfs(self: AvdStructuredConfigBaseProtocol, snmp_settings: EosDesigns.SnmpSettings) -> None:
         """
         Set list of dicts for enabling/disabling SNMP for VRFs.
 
         Requires one of the following options to be set under snmp_settings:
         - vrfs
-        - enable_mgmt_interface_vrf
-        - enable_inband_mgmt_vrf
+        - enable_mgmt_interface_vrf     TODO: 6.0 remove
+        - enable_inband_mgmt_vrf        TODO: 6.0 remove
         """
         has_mgmt_ip = (self.shared_utils.node_config.mgmt_ip is not None) or (self.shared_utils.node_config.ipv6_mgmt_ip is not None)
 
-        vrfs = snmp_settings.vrfs._deepcopy()
+        vrfs = EosCliConfigGen.SnmpServer.Vrfs()
+
+        # TODO: 6.0 remove the if condition.
+        if self.inputs.avd_6_behaviors.snmp_settings_vrfs:
+            vrfs.append_new(name="default", enable=False)
+
+        for vrf in snmp_settings.vrfs:
+            if vrf.enable is None:
```

Comment:
we don't have an example of missing enable so it is not covered. We could either: make `enable` required & remove the check or add a VRF in one example

Reply:
enable should not be required. You may only use the model to specify source-interface